src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java

```diff
@@ -440,23 +435,27 @@ public boolean shouldSkipMonitoredTripCheck() throws Exception {
         // get the configured timezone that OTP is using to parse dates and times
         ZoneId targetZoneId = DateTimeUtils.getOtpZoneId();
 
-        // Make sure journey state indicates that the trip should still possible on at least one day of the week as of
-        // the last time a matching itinerary was attempted to be found. This field in the journey state is used instead
-        // of the trip's itinerary existence data in order to not skip trips that are monitored only one day of the week
-        // and may have not had a matching itinerary on that single monitored day even though it could be possible the
-        // next week.
-        if (previousJourneyState.noLongerPossible) return true;
+        // If trip is no longer possible, no further checking is needed. The itinerary existence data should not be
+        // checked here to avoid incorrectly skipping trips that are monitored on a single day of the week, but which
+        // may have not had a matching itinerary on that day for one week (even though the trip could be possible the
+        // next week).
+        if (
+            previousJourneyState.tripStatus != null &&
```

Comment:
Is the null check necessary? I think we can just use `==` for better readability.

Reply:
Refactored as suggested in d95910c71758df3212e5f878a91a509f02839dd0.